math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -537,6 +537,18 @@ void UninitializedValueConstruct(ForwardIt first, ForwardIt last)
 #endif
 }
 
+/// An unsafe function to reset the buffer for which this RVec is acting as a view.
+///
+/// \note This is a low-level method that _must_ be called on RVecs that are already non-owning:
+/// - it does not put the RVec in "non-owning mode" (fCapacity == -1)
```

Comment:
I think we should add at least an assert that this is correct ... but this is in a header file ... so not great ...

Reply:
As per the commit message, this function serves as a performance optimization. It is in `ROOT::Internal` and it is clearly documented as unsafe. 

I think it's ok if it has sharp edges, it is its purpose to be a sharp tool.

Reply:
(on the same line, I could probably have it take an RVecImpl and a void* and put it in a source file where it's safer to add an `assert`, but it's actually better to keep the function in the header so it's simpler for the compiler to inline it)